src/compiler.ts

```diff
@@ -6241,6 +6247,7 @@ export class Compiler extends DiagnosticEmitter {
         this.makeReplace(
           valueExpr,
           module.global_get(global.internalName, nativeType),
+          type,
```

Comment:
This is similar to my comment above, except that we don't even have `valueType` here, hmm. Perhaps it makes sense to add `valueType` to the signature of `makeGlobalAssignment` to align it with the one above?